velox/functions/prestosql/tests/RandTest.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+
+class RandTest : public functions::test::FunctionBaseTest {
+ protected:
+  static constexpr vector_size_t size = 1'000;
+  template <typename T>
+  facebook::velox::RowVectorPtr generateRowVector(T data) {
+    return makeRowVector(
+        {makeFlatVector<T>(size, [=](auto /*row*/) { return data; })});
+  }
+};
+
+namespace {
+
+TEST_F(RandTest, zeroArg) {
+  auto result = evaluate<SimpleVector<double>>(
+      "random()", generateRowVector<int32_t>(123));
+  for (int32_t i = 0; i < result->size(); ++i) {
+    ASSERT_TRUE(result->valueAt(i) < 1.0);
+    ASSERT_TRUE(result->valueAt(i) >= 0.0);
+  }
+}
+
+TEST_F(RandTest, negativeInt32) {
+  VELOX_ASSERT_THROW(
```

Comment:
For each case that throws, add a matching case using TRY to very it doesn't throw an return null.